core_impl/src/main/java/io/opencensus/stats/MutableViewData.java

```diff
@@ -136,9 +140,45 @@ ViewData toViewData(Clock clock) {
       }
       map.put(entry.getKey(), aggregates);
     }
-    return ViewData.create(view, map, CumulativeData.create(start, clock.now()));
+
+    WindowData windowData = view.getWindow().match(
+        new CreateCumulativeData(start, clock.now()),
+        new CreateIntervalData(clock.now()),
+        Functions.<WindowData>throwIllegalArgumentException());
```

Comment:
I think AssertionError would be better if we expect the API and implementation to always have the same version.  Alternatively, we could try to handle an unknown window, possibly by treating it as cumulative.